radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/CandidateForkVotesPostProcessor.java

```diff
@@ -0,0 +1,192 @@
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.statecomputer;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.hash.HashCode;
+import com.radixdlt.application.validators.state.ValidatorSystemMetadata;
+import com.radixdlt.atom.SubstateTypeId;
+import com.radixdlt.consensus.bft.BFTNode;
+import com.radixdlt.consensus.bft.BFTValidatorSet;
+import com.radixdlt.constraintmachine.REProcessedTxn;
+import com.radixdlt.constraintmachine.RawSubstateBytes;
+import com.radixdlt.constraintmachine.SubstateDeserialization;
+import com.radixdlt.constraintmachine.SubstateIndex;
+import com.radixdlt.engine.PostProcessor;
+import com.radixdlt.engine.PostProcessorException;
+import com.radixdlt.serialization.DeserializeException;
+import com.radixdlt.statecomputer.forks.CandidateForkVote;
+import com.radixdlt.statecomputer.forks.ForkVotingResult;
+import com.radixdlt.store.EngineStore;
+import com.radixdlt.utils.Bytes;
+import com.radixdlt.utils.Pair;
+import com.radixdlt.utils.UInt256;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/** Adds forksVotingResults at epoch boundary to the result metadata. */
+public final class CandidateForkVotesPostProcessor implements PostProcessor<LedgerAndBFTProof> {
+  /**
+   * Specifies a minimum stake that needs to vote for a fork to be added to the resulting stored
+   * map. Acts as a storage size optimization so that we don't store insignificant entries.
+   */
+  private static final short VOTES_THRESHOLD_TO_STORE_RESULT = 5000; /* 50.00% */
+
+  private final SubstateDeserialization substateDeserialization;
+
+  public CandidateForkVotesPostProcessor(SubstateDeserialization substateDeserialization) {
+    this.substateDeserialization = Objects.requireNonNull(substateDeserialization);
+  }
+
+  @Override
+  public LedgerAndBFTProof process(
+      LedgerAndBFTProof metadata,
+      EngineStore.EngineStoreInTransaction<LedgerAndBFTProof> engineStore,
+      List<REProcessedTxn> txns)
+      throws PostProcessorException {
+    if (metadata.getProof().getNextValidatorSet().isPresent()) {
+      return metadata.withForksVotingResults(prepareForksVotingResults(engineStore, metadata));
+    } else {
+      return metadata;
+    }
+  }
+
+  private ImmutableSet<ForkVotingResult> prepareForksVotingResults(
+      EngineStore.EngineStoreInTransaction<LedgerAndBFTProof> engineStore,
+      LedgerAndBFTProof ledgerAndBFTProof) {
+    final var nextEpoch = ledgerAndBFTProof.getProof().getEpoch() + 1;
+    final var validatorSet = ledgerAndBFTProof.getProof().getNextValidatorSet().orElseThrow();
+    final var totalPower = validatorSet.getTotalPower();
```

Comment:
What is total power? Do we have a power concept in the system?

Reply:
yup, an abstraction for stake, see `BFTValidatorSet`

Reply:
are they synonyms?

Reply:
To some extent but they're on different abstraction levels. The consensus layer doesn't really know that power is backed by stake

Reply:
I was wondering if we could just use "stake" in both cases.

Reply:
why?

Reply:
Cause stake is a well defined concept in the DLT domain.